demo/app/android/app/src/androidTest/kotlin/integration_test.kt

```diff
@@ -137,13 +138,48 @@ class IntegrationTest {
         println("requestURI ->")
         println( requestURI)
 
-        val requiredOpenID4CIArgs = IssuerInitiatedInteractionArgs(requestURI, crypto, didResolver)
+        val requiredOpenID4CIArgs = InteractionArgs(requestURI, crypto, didResolver)
         println("requiredOpenID4CIArgs")
         println(requiredOpenID4CIArgs)
         val ciOpts = InteractionOpts()
         ciOpts.addHeader(trace.traceHeader())
 
-        val ciInteraction = IssuerInitiatedInteraction(requiredOpenID4CIArgs, ciOpts)
+        val ciInteraction = Interaction(requiredOpenID4CIArgs, ciOpts)
+        var clientID = "oidc4vc_client"
+        val redirectURI = "http://127.0.0.1/callback"
+
+        assertThat(ciInteraction.dynamicClientRegistrationSupported()).isTrue()
+
+        if (ciInteraction.dynamicClientRegistrationSupported()){
+            var dynamicRegistrationEndpoint = ciInteraction.dynamicClientRegistrationEndpoint()
+            assertThat(dynamicRegistrationEndpoint).isNotEmpty()
+
+            var clientMetadata = Oauth2.newClientMetadata()
+            var grantTypesArr = StringArray()
+            grantTypesArr.append("authorization_code")
+            clientMetadata.setGrantTypes(grantTypesArr)
+            assertThat(clientMetadata.grantTypes()).isNotNull()
+
+            var redirectUri = StringArray()
+            redirectUri.append(redirectURI)
+            clientMetadata.setRedirectURIs(redirectUri)
+            assertThat(clientMetadata.redirectURIs()).isNotNull()
+
+
+            clientMetadata.setScope("openid profile")
+            clientMetadata.setTokenEndpointAuthMethod("none")
+
+            var authorizationCodeGrantParams = ciInteraction.authorizationCodeGrantParams()
+            if (authorizationCodeGrantParams.hasIssuerState()){
+                var issuerState = authorizationCodeGrantParams.issuerState()
+                clientMetadata.setIssuerState(issuerState)
+                assertThat(clientMetadata.issuerState()).isNotEmpty()
+            }
+
+            var registrationResp = Oauth2.registerClient(dynamicRegistrationEndpoint, clientMetadata, null)
```

Comment:
You'll want to pass through the scopes from the response like I did in my latest PR (just merged in)